### PR TITLE
[SPARK-29872] Improper cache strategy in examples

### PR DESCRIPTION
**Pull request by Icysandwich**
### What changes were proposed in this pull request?
Correct some cache strategy in examples.

### Why are the changes needed?
These changes can improve the performance of examples.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Manually

### CODE REVIEW

File: examples/src/main/scala/org/apache/spark/examples/ml/LogisticRegressionSummaryExample.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ object LogisticRegressionSummaryExample {
     println(s"areaUnderROC: ${trainingSummary.areaUnderROC}")
 
     // Set the model threshold to maximize F-Measure
-    val fMeasure = trainingSummary.fMeasureByThreshold
+    val fMeasure = trainingSummary.fMeasureByThreshold.cache()
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/ml/LogisticRegressionSummaryExample.scala]: I'm not sure this one is important for an example. It won't matter much

File: examples/src/main/scala/org/apache/spark/examples/SparkTC.scala [hunk 2]
```diff
@@ -55,20 +55,24 @@ object SparkTC {
     // the graph to obtain the path (x, z).
 
     // Because join() joins on keys, the edges are stored in reversed order.
-    val edges = tc.map(x => (x._2, x._1))
+    val edges = tc.map(x => (x._2, x._1)).cache()
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/SparkTC.scala]: This is more complex for an example, but, I think it's reasonably important to show this pattern in this type of example.

File: examples/src/main/scala/org/apache/spark/examples/sql/SparkSQLExample.scala [hunk 3]
```diff
@@ -198,6 +198,7 @@ object SparkSQLExample {
       .textFile("examples/src/main/resources/people.txt")
       .map(_.split(","))
       .map(attributes => Person(attributes(0), attributes(1).trim.toInt))
+      .cache()
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/sql/SparkSQLExample.scala]: This one isn't important enough to bother with, I think. The data is small

**Review by srowen — CHANGES_REQUESTED** [review]

**Comment by srowen** [comment]
BTW "improper" isn't quite the right word. These aren't bugs per se.

**Comment by Icysandwich** [comment]
Okay, I see.

**Comment by srowen** [comment]
I think the SparkTC change was OK, but yeah the examples just don't matter much. Maybe it's best to not clutter the example.

**Pull request closed without merging**